code/src/components/ToDoInput.js

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from "react";
+import { useDispatch } from "react-redux";
+import { todos } from "../reducers/todos.js";
+import styled from "styled-components";
+
+export const TodoInput = ({ listId }) => {
+  const dispatch = useDispatch();
+  const [inputValue, setInputValue] = useState("");
+
+  const handleOnSubmit = (e) => {
+    e.preventDefault();
+
+    dispatch(
+      todos.actions.addTodo({
+        listId: listId,
+        itemInfo: { description: inputValue, done: false },
+      })
+    );
```

Comment:
It's possible to submit an empty todo now, maybe check if the input is empty or not before dispatching, or setting the input-field to be required?